mqtt/mqtt-broker/src/session.rs

```diff
@@ -910,6 +907,103 @@ impl Default for PacketIdentifiers {
     }
 }
 
+/// `BoundedQueue` is a queue of publications with bounds by count and total payload size in bytes.
+///
+/// Packets will be queued until either of `max_len` or `max_size` limits is reached, and
+/// then `when_full` strategy is applied.
+///
+/// None for `max_len` or `max_size` means "unbounded".
+#[derive(Clone, Debug, PartialEq)]
+struct BoundedQueue {
```

Comment:
I think this code deserves separate module and more tests in isolation to cover corner cases

Reply:
Maybe, but I was not trying to make a publicly available general purpose collection, rather just refactor some code a bit. I would keep it as internal details of `SessionState` and would rather move SessionState+BoundedQueue into a separate module.

Reply:
Do you think we have enough tests to cover queue limits?

Reply:
I think I've covered all cases I could think of, but they cover `publish_to` public method that in turn uses private `BoundedQueue`.